addons/website_forum/populate/mail_message.py

```diff
@@ -0,0 +1,81 @@
+    _populate_sizes = {'small': 100, 'medium': 1000, 'large': 90000}
+
+    def _populate_factories(self):
+        comment_subtype = self.env.ref('mail.mt_comment')
+
+        def get_res_id_and_create_date(iterator, *args):
+            """Randomly assign messages to some populated posts and answers.
+            This makes sure these questions and answers get one to four comments.
+            Also define a create_date that is more recent than the post/answer's create_date
+            """
+            question_ids = self.env.registry.populated_models['forum.post']
+            posts = self.env['forum.post'].search_read(['|', ('id', 'in', question_ids), ('parent_id', 'in', question_ids)], ['id', 'create_date'])
+            sample_size = int(len(question_ids) * 0.7)
+            res_ids = populate.random.sample(posts, sample_size)
+            nb_comments = 0
+            idx = 0
+            post_to_comment_id = None
+            create_date = None
+            hours = [_ for _ in range(1, 24)]
+
+            for values in iterator:
+                if nb_comments == 0:
+                    nb_comments = populate.random.choices(*zip(*CP_WEIGHTS.items()))[0]
+                    post_to_comment_id = res_ids[idx]['id']
+                    idx += 1
+                nb_comments -= 1
+                create_date = fields.Datetime.add(res_ids[idx]['create_date'], hours=populate.random.choice(hours))
+                yield {**values, 'res_id': post_to_comment_id, 'create_date': create_date}
+
+        def get_author_and_create_ids(iterator, *args):
+            user_ids = self.env.registry.populated_models['res.users']
+            author_and_user_ids = self.env['res.users'].search_read([('id', 'in', user_ids)], ['id', 'partner_id'])
+
+            for values in iterator:
+                ids = populate.random.choice(author_and_user_ids)
+                yield {**values, 'author_id': ids['partner_id'][0], 'create_uid': ids['id']}
+
+        def get_write_date(values=None, random=None, **kwargs):
+            return values['create_date']
+
+        def get_write_uid(values=None, random=None, **kwargs):
+            return values['create_uid']
+
+        return [
+            ('_author_and_create_ids', get_author_and_create_ids),
+            ('body', populate.constant('message_body_{counter}')),
+            ('message_type', populate.constant('comment')),
+            ('model', populate.constant('forum.post')),
+            ('_res_id_and_create_date', get_res_id_and_create_date),
+            ('subtype_id', populate.constant(comment_subtype.id)),
+            ('write_date', populate.compute(get_write_date)),
+            ('write_uid', populate.compute(get_write_uid)),
+        ]
+
+    def _populate(self, size):
+        records = super()._populate(size)
+
+        _logger.info('forum.post: update last_activity_date for posts with comments')
+        comments = self._read_group([('id', 'in', records.ids)], ['res_id'], ['create_date:max'])
```

Comment:
Doesn't this leave out the comments on answers?

How about making a mapping of comment res_id to create date, then iterate over each **question** `q` and set `q.last_activity_date` to  the max of `q.answer_ids.mapped(lambda a: comments_to_date.get(a.id, q.last_activity_date))`?